pysatNASA/instruments/methods/cdaweb.py

```diff
@@ -7,18 +7,342 @@
 
 import datetime as dt
 import os
-import requests
 import sys
+import re
+import requests
+import numpy as np
+import cdflib
 
 from bs4 import BeautifulSoup
 import pandas as pds
 
-import pysatCDF
+import pysat
 from pysat import logger
 from pysat.utils import files as futils
 from pysat.instruments.methods import general
 
 
+def convert_ndimensional(data, index=None, columns=None):
+    """converts high-dimensional data to a Dataframe"""
+    if columns is None:
+        columns = [range(i) for i in data.shape[1:]]
+        columns = pds.MultiIndex.from_product(columns)
+
+    return pds.DataFrame(data.T.reshape(data.shape[0], -1),
+                         columns=columns, index=index)
+
+
+class CDF():
+    """cdflib wrapper for loading time series data
+
+    Loading routines borrow heavily from pyspedas's cdf_to_tplot function
+    """
+
+    def __init__(self, filename,
+                 varformat='*',  # regular expressions
+                 var_types=['data', 'support_data'],
+                 center_measurement=False,
+                 raise_errors=False,
+                 regnames=None,
+                 datetime=True,
+                 **kwargs):
+        self._raise_errors = raise_errors
+        self._filename = filename
+        self._varformat = varformat
+        self._var_types = var_types
+        self._datetime = datetime
+        self._var_types = var_types
+        self._center_measurement = center_measurement
+
+        # registration names map from file params to kamodo-compatible names
+        if regnames is None:
+            regnames = {}
+        self._regnames = regnames
+
+        self._cdf_file = cdflib.CDF(self._filename)
+        self._cdf_info = self._cdf_file.cdf_info()
+        self.data = {}  # python-in-Heliophysics Community data standard
+        self.meta = {}  # python-in-Heliophysics Community metadata standard
```

Comment:
It would be cleaner to specify this as a block comment above these two definitions, and also supply the location for this standard.

Reply:
Is this a pyhc standard, or is this a pysat standard?

Reply:
It's not a pysat standard.  Maybe @jklenzing has more knowledge?